public/components/event_analytics/explorer/context/context.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useContext, createContext, createElement } from 'react';
+import { createNotifications } from '../../../../../../../src/plugins/opensearch_dashboards_react/public';
+import {
+  DashboardsObservabilityContext,
+  OpenSearchDashboardsServices,
+  ObservabilityDashboardsReactContextValue,
+  DashboardsObservabilityServices,
+} from './types';
+import { useMemo } from 'react';
+import React from 'react';
+
+const defaultObservabilityContextValue = {
+  services: {} as DashboardsObservabilityServices,
+  notifications: createNotifications({}),
+};
+
+export const observabilityContext = createContext<
+  ObservabilityDashboardsReactContextValue<DashboardsObservabilityServices>
+>(defaultObservabilityContextValue);
+
+export const useDashboardsObservability = (): ObservabilityDashboardsReactContextValue<
+  DashboardsObservabilityServices
+> => useContext(observabilityContext);
+
+export const createDashboardsObservabilityContext = <Services extends OpenSearchDashboardsServices>(
+  services: Services
+): DashboardsObservabilityContext<Services> => {
+  // common context
+  const value = {
+    services,
+    notifications: createNotifications(services),
+  };
+
+  const Provider: React.FC<{ services?: Services; children: React.ReactDOM }> = ({
+    services: additionalServices = {},
+    children,
+  }) => {
+    const oldValue = useDashboardsObservability();
+    const { value: mergedValue } = useMemo(
+      () =>
+        createDashboardsObservabilityContext({
```

Comment:
`Services` is a generic type that extends `OpenSearchDashboardsServices`, how does it ensure that Observability specific services like `pplService` are passed in?